tests/backup/backup_helper.go

```diff
@@ -5559,3 +5678,242 @@ func UpgradeKubevirt(versionToUpgrade string, workloadUpgrade bool) error {
 	}
 	return nil
 }
+
+// ChangeAdminNamespace changes admin namespace for Storage Cluster
+func ChangeAdminNamespace(namespace string) (*v1.StorageCluster, error) {
+	// Get current storage cluster configuration
+	isOpBased, err := Inst().V.IsOperatorBasedInstall()
+	if err != nil {
+		return nil, err
+	}
+	storkDeploymentNamespace, err := k8sutils.GetStorkPodNamespace()
+	storkOldPods, err := core.Instance().GetPods(storkDeploymentNamespace, map[string]string{"name": "stork"})
+	if err != nil {
+		return nil, err
+	}
+	stc, err := Inst().V.GetDriver()
+	if err != nil {
+		return nil, err
+	}
+	log.Infof("Is op based deployment %v", isOpBased)
+
+	if isOpBased {
+		if adminNamespace, ok := stc.Spec.Stork.Args["admin-namespace"]; ok {
+			log.Infof("Current admin namespace - [%s]", adminNamespace)
+		}
+		// Setting the new admin namespace
+		if namespace != "" {
+			stc.Spec.Stork.Args["admin-namespace"] = namespace
+		} else {
+			delete(stc.Spec.Stork.Args, "admin-namespace")
+		}
+		stc, err = operator.Instance().UpdateStorageCluster(stc)
+		if err != nil {
+			return nil, err
+		}
+		if namespace != "" {
+			log.Infof("Configured admin namespace to %s", namespace)
+		} else {
+			log.Infof("Removed admin namespace") // Removing admin namespace is not supported - https://docs.portworx.com/portworx-backup-on-prem/configure/admin-namespace.html
+		}
+	} else {
+		log.Infof("Updating stork deployment as it's pxe is not present")
+		storkDeployment, err := apps.Instance().GetDeployment(storkDeploymentName, storkDeploymentNamespace)
+		if err != nil {
+			return nil, err
+		}
+		storkDeployment.Spec.Template.Spec.Containers[0].Command = append(storkDeployment.Spec.Template.Spec.Containers[0].Command, fmt.Sprintf("--admin-namespace=%s", namespace))
+
+		_, err = apps.Instance().UpdateDeployment(storkDeployment)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	updatedStorkDeployment, err := apps.Instance().GetDeployment(storkDeploymentName, storkDeploymentNamespace)
+	if err != nil {
+		return nil, err
+	}
+	err = apps.Instance().ValidateDeployment(updatedStorkDeployment, storkPodReadyTimeout, podReadyRetryTime)
+	if err != nil {
+		return nil, err
+	}
+
+	// Removing all old stork pods
+	for _, pod := range storkOldPods.Items {
+		err = core.Instance().DeletePod(pod.Name, pod.Namespace, false)
+		if err != nil {
+			log.Warnf("Unable to delete [%v]. Error - [%s]", pod.Name, err.Error())
+		} else {
+			log.Infof("Old stork pod deleted [%v]", pod.Name)
+		}
```

Comment:
You can use this function instead `func DeletePodWithLabelInNamespace(namespace string, label map[string]string) error`

Reply:
We are only deleting old stork pods(Running before admin namespace update), DeletePodWithLabelInNamespace won't be able to identify if the pod is new or old and will delete all pods